scripts/globals/besieged.lua

```diff
@@ -1,217 +1,159 @@
-require("scripts/globals/keyitems");
+-----------------------------------
+--
+--     Functions for Besieged system
+--
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/npc_util")
+require("scripts/globals/status")
+-----------------------------------
+
+dsp = dsp or {}
+dsp.besieged = dsp.besieged or {}
+
+dsp.besieged.onTrigger = function(player, npc, eventBase)
+    local mercRank = getMercenaryRank(player)
+    if mercRank == 0 then
+        player:startEvent(eventBase + 1, npc)
+    else
+        local maps = bit.bor(getMapBitmask(player), bit.lshift(getAstralCandescence(), 31))
+        player:startEvent(eventBase, player:getCurrency("imperial_standing"), maps, mercRank, 0, unpack(getImperialDefenseStats()))
+    end
+end
+
+dsp.besieged.onEventUpdate = function(player, csid, option)
+    local itemId = getISPItem(option)
+    if itemId and option < 1073741824 then
+        local maps = bit.bor(getMapBitmask(player), bit.lshift(getAstralCandescence(), 31))
+        player:updateEvent(player:getCurrency("imperial_standing"), maps, getMercenaryRank(player), player:canEquipItem(itemId) and 2 or 1, unpack(getImperialDefenseStats()))
+    end
+end
+
+dsp.besieged.onEventFinish = function(player, csid, option)
+    local ID = zones[player:getZoneID()]
+    if option == 0 or option == 16 or option == 32 or option == 48 then
+        -- Sanction
+        if option ~= 0 then
+            player:delCurrency("imperial_standing", 100)
+        end
+        
+        player:delStatusEffect(dsp.effect.SIGIL)
+        player:delStatusEffect(dsp.effect.SANCTION)
+        player:delStatusEffect(dsp.effect.SIGNET)
```

Comment:
Looking at clips of retail, these should be silently deleted.

Also, a flag to remove all influence buffs at once would be nice, instead of needing 4 `delStatusEffect` for everything that uses them.